mkt/files/models.py

```diff
@@ -103,7 +105,7 @@ def addon(self):
     def generate_hash(self, filename=None):
         """Generate a hash for a file."""
         hash = hashlib.sha256()
-        with storage.open(filename or self.file_path, 'rb') as obj:
+        with private_storage.open(filename or self.file_path, 'rb') as obj:
```

Comment:
I think I need to look more closely at this. If the `File` is public this will be incorrect. Assuming the `mv` stuff below works as it should we can toggle the storage backend based on `file_ == mkt.STATUS_DISABLED`.